agent/healthcheck/health_check.go

```diff
@@ -196,8 +199,19 @@ func (healthStatus *HealthStatus) computeHealthCheck(agentConfig config.AgentCon
 	// it is reported as Unhealthy.
 	// Disconnection from Control plane (Applicable only when not using Relay Mode):
 	// - Even when initialized, Envoy is statically stable and hence continue to report healthy.
-	// - To prevent Envoy from getting stuck in a disconnected state, we define a configurable threshold crossing which
-	//   it starts reporting unhealthy.
+    // - If the connection status is CONNECTED, Envoy is reported as Healthy, and FlipTimestamps is reset.
+    // - If the connection status is NOT CONNECTED and has changed since the last check,
+    //   the current time is appended to FlipTimestamps, provided that the timestamp is within the last 30 minutes.
+    // - If the length of FlipTimestamps reaches or exceeds a predefined threshold of 10, Envoy is reported as Unhealthy.
```

Comment:
What if the task on EC2 gets terminated and placed on the same instance again where the Relay is problematic.